src/main/java/gregtech/api/util/CTRecipeHelper.java

```diff
@@ -43,7 +46,46 @@ public static String getMetaItemId(ItemStack item) {
                 return ((BlockMaterialPipe<?, ?, ?>) block).getPrefix().name + ((BlockMaterialPipe<?, ?, ?>) block).getItemMaterial(item).toCamelCaseString();
             }
         }
+        return null;
+    }
 
+    @Nullable
+    public static Material getMaterial(ItemStack item) {
```

Comment:
Why not use `OreDictUnifier#getMaterial(ItemStack)`?

Reply:
Because i didn't know it exists